torch/csrc/jit/runtime/graph_executor.cpp

```diff
@@ -299,7 +300,12 @@ struct DifferentiableGraphBackward : public autograd::Node {
         addOutputForTensor(tensor);
       }
     } else {
-      addOutputForTensor(value.toTensor());
+      if (value.isTensor()) {
+        addOutputForTensor(value.toTensor());
+      } else {
+        // We could have None passed here via `Optional[Tensor]`
+        add_next_edge(autograd::Edge{});
```

Comment:
this lgtm,  but ping @albanD just to be safe

Reply:
That looks good. Especially because it matches what is done in `addOutputForTensor` just above.